haskell/src/Stablecoin/Client/Impl.hs

```diff
@@ -327,12 +316,55 @@ getMintingAllowance contract minter = do
   let allowanceMaybe = M.lookup minterAddr mintingAllowances
   pure $ fromMaybe 0 allowanceMaybe
 
--- | Get the token metadata of the contract
 getTokenMetadata :: "contract" :! AddressOrAlias -> MorleyClientM FA2.TokenMetadata
 getTokenMetadata contract = do
-  mdRegistry <- sTokenMetadataRegistry <$> getStorage contract
-  bigMapId <- mrsTokenMetadata <$> getRegistryStorage mdRegistry
-  readBigMapValue bigMapId FA2.theTokenId
+  (metadata, storageView) <- getContractMetadata contract
+  case TZ.getViews metadata of
+    Left err -> throwM $ SCEMetadataError ("Views was not found in metadata:" <> show err)
+    Right views -> case MD.findView @(ToT Storage) views "GetTokenMetadata" of
+      Nothing -> throwM $ SCEMetadataError "'GetTokenMetadata' view was not found in metadata"
+      Just getTMD -> do
+        let storageWithEmptyBm :: Storage =
+              -- In the below code we convert the `StorageView` to `Storage` by
+              -- replacing all bigmaps with empty counterparts, and this should be fine here since
+              -- the token metadata view does not access any of these big maps.
+              storageView
+                { sLedger = mempty
+                , sMetadata = mempty
+                , sOperators = mempty
+                , sPermits = mempty
+                }
+        case MD.interpretView @(Natural, Map MText ByteString)
+            dummyContractEnv getTMD (MD.ViewParam (0 :: Natural)) storageWithEmptyBm of
+          Right (_, md) -> pure md
+          Left err -> throwM $ SCEMetadataError ("Error while interpreting the contract:" <> show err)
+
+-- | Get the metadata of the contract
+getContractMetadata
+  :: "contract" :! AddressOrAlias
+  -> MorleyClientM (TZ.Metadata (ToT Storage), StorageView)
+getContractMetadata contract = do
+  storageView <- getStorage contract
+  let bigMapId = sMetadata storageView
+  metadataUri <- decodeUtf8 <$> readBigMapValue bigMapId [mt||]
+  case parseMetadataUri metadataUri of
+    Nothing -> throwM $ SCEMetadataError ("Unparsable URI:" <> metadataUri)
+    Just uri -> case uri of
```

Comment:
`throwMdErr` could be used here as well with something like:
```suggestion
  uri <- throwMdErr id . maybeToRight ("Unparsable URI:" <> metadataUri) $ parseMetadataUri metadataUri
  case uri of
```
or even better I'd suggest to modify `parseMetadataUri` to be:
```haskell
parseMetadataUri :: Text -> Either Text ParsedMetadataUri
parseMetadataUri = first (fromString . errorBundlePretty) . P.parse "" metadataUriParser
```
and here just have:
```suggestion
  uri <- throwMdErr ("Unparsable URI:" <>) $ parseMetadataUri metadataUri
  case uri of
```